avionics/common/include/states/ascent_to_apogee.h

```diff
@@ -30,21 +31,21 @@ class AscentToApogee : public IState {
      */
     StateId getNewState(Calculator const &input) {
         if (input.velocityGroundZ() > MACH_LOCK_VELOCITY_) {
-            if (++mach_checks >= MACH_LOCK_CHECKS_) {
-                return mach_lock_id_;
+            if (++mach_checks_ >= MACH_LOCK_CHECKS_) {
+                return machTimeout();
```

Comment:
I don't think this does what it should. 

The intended MachLock timeout is that we want a maximum time to stay in MachLock for; this means that the MachLock state needs to be edited.

Once the state enters MachLock, it'll be checking the MackLock object for when to exit - not ascent to apogee.

Reply:
Oops, that's my bad I misunderstood the code. I've fixed it in my latest commit.